templates/tfengine/components/cicd/configs/tf-apply.yaml

```diff
@@ -38,3 +39,6 @@ steps:
     args: ["./cicd/configs/run.sh", "-d", "${_MANAGED_DIRS}", "-a", "init", "-a", "plan", "-a", "apply -auto-approve"]
     dir: "${_TERRAFORM_ROOT}"
     id: Apply
+
+options:
+  workerPool: "${_WORKER_POOL}"
```

Comment:
Shouldn't this just be `pool` and be an object? See the example here: https://cloud.google.com/build/docs/private-pools/run-builds-in-private-pool

Reply:
I tested this. Using `pool: name:` and `workerPool` lead to the same results.

Reply:
Interesting. Could you find documentation for this variation then? I think if in the future someone Googles this they might end up on the same docs as me and wonder why it's working.

Reply:
https://cloud.google.com/build/docs/api/reference/rest/v1/projects.builds#Build.BuildOptions

seems `workerPool` is a legacy field and is deprecated. `pool` is the new one.

Reply:
Ah so my suspicion was valid.

Reply:
This terraform doc indicates `worker-pool` as a valid argument for `google_cloudbuild_trigger` `build-trigger`. However, it's doesn't specifically include the valid arguments for yaml files. I assume the arguments should have a parity.

Reply:
Ah I see! Nice find!

Reply:
The official gcp API doc should be treated as the SoT. Terraform resources fields/docs are derived from those and can be out-of-date more easily. The `workerPool` field in https://cloud.google.com/build/docs/api/reference/rest/v1/projects.builds#Build.BuildOptions is indicated as deprecated.

Reply:
Updated the to `pool.name`